### ISSUE TITLE
"Sort from smallest file to largest" button or option

### ISSUE THREAD

**Issue by StabbyPancake** [Radarr/Radarr#3338]
It would be great if we had a button for "Sort from smallest file size to largest" somewhere in the settings.  By design, Radarr sorts from largest file size down to smallest.  While this works for many people who have nearly unlimited space, there are quite a few of us out there that prefer compact rips in order to maximize our space, and folks that are on monthly download caps.  

If there is a 15GB rip and a 6GB rip, and both are in the BluRay-1080p category, I would GREATLY prefer radarr grab the 6GB rip.  Unfortunately, the only real way to do this currently is through complicated custom formats like this:  https://imgur.com/tGwKvvc

I've basically got this worked out using custom formats, but it's fairly involved, and I know there are quite a few people that have seen my configuration that would appreciate a simple way to do this.  While there is a place in this world for 27GB BluRay rips, there are also those of us who would prefer a smaller file. 

The interface:
A checkbox somewhere in the configuration labeled something like "Sort available downloads from smallest to largest".  File size minimums and maximums for each category are already well defined by using the cutoff sliders, so there is no need to implement anything further there, we just need a way to sort smallest to largest.

**Comment by fryfrog** [comment]
I swear there is another issue or feathub request for this already, could you dig for it and link it here and then close this one? I'm pretty sure I've seen talk around an "ideal size" solution to this.

**Comment by StabbyPancake** [comment]
@fryfrog Read the exchange between myself and Gallileo from a few days ago on Discord.  I do NOT want an "ideal size" feature, I simply want the ability to invert the file size sort order so that it searches from smallest to largest.  As I explained, the "ideal" size for an action movie with lots of graphics and explosions will not be the ideal size for a comedy or animated movie that gets requested in the same quality (i.e. BluRay-720p).  Ideal size sounds like a nice addition, but I believe this is a truly separate feature that could be (from the perspective of a non-programmer such as myself) easily added to Radarr without too much hassle or changes to the code.

**Comment by fryfrog** [comment]
Ah, that must be where I was remembering. But I seriously doubt your reverse sort request is new, did you go digging for an existing issue or feathub request? Or are you going to make me do it? :)

**Comment by StabbyPancake** [comment]
This is the request you are referencing:  https://github.com/Radarr/Radarr/issues/1224

But, as I explained, this is a different function than what I am asking for.  Let me give you a for instance:
1.  I set my preferred range from 3GB-4GB, with a maximum file size of 14GB.  
2.  Radarr searches for my movie, but doesn't find anything in the "preferred" range, so it waits the allotted time before searching for a different release
3.  SURPRISE!  Radarr is STILL defaulted to search for largest file size first, so since it has waited X days when it couldn't find my preferred size, not it defaults back to the normal search order, and I get a 13GB file, when there was a 5GB file already available.  Since it wasn't in my "preferred" range though, it skips it, and goes through releases from largest to smallest until it snags a download (which in this example would be the 13GB one).

That's why I say:  These are 2 separate requests, meant to do 2 separate things.

I did read through the existing requests and didn't see anything that asked for what I was looking for.
